main/accounts/index.js

```diff
@@ -565,17 +565,81 @@ class Accounts extends EventEmitter {
     delete this.accounts[address]
   }
 
-  setGasPrice (price, handlerId, cb) {
-    if (!price || isNaN(parseInt(price, 'hex')) || parseInt(price, 'hex') < 0) return cb(new Error('Invalid price'))
+  invalidValue (fee) {
+    return (!fee || isNaN(parseInt(fee, 'hex')) || parseInt(fee, 'hex') < 0)
+  }
+
+  setBaseFee (baseFee, handlerId, cb) {
+    if (this.invalidValue(baseFee)) return cb(new Error('Invalid base fee'))
+    if (!this.current()) return
+    if (this.current().requests[handlerId] && this.current().requests[handlerId].type === 'transaction') {
+      if (parseInt(baseFee, 'hex') > 9999 * 1e9) baseFee = '0x' + (9999 * 1e9).toString(16)
+      const priorityFee  = this.current().requests[handlerId].data.maxPriorityFeePerGas
+      const gasLimit = this.current().requests[handlerId].data.gasLimit
+      const fee = parseInt(baseFee, 'hex') + parseInt(priorityFee, 'hex')
+      const limit = parseInt(gasLimit, 'hex')
+      if (fee * limit > FEE_MAX) {
+        log.warn('Operation would set fee over hard limit')
+        baseFee = '0x' + (Math.floor(FEE_MAX / limit) - parseInt(priorityFee, 'hex')).toString(16)
+      }
+      this.current().requests[handlerId].data.maxFeePerGas = baseFee
```

Comment:
is this logic right? we're passing in just the base fee but we're setting the overall max fee which includes both the base fee and the priority fee

Reply:
We grab the current priorityFee on line 577

Reply:
but do we do anything with it? maybe I'm misinterpreting this but this line looks like it ignores that: `this.current().requests[handlerId].data.maxFeePerGas = baseFee`

Reply:
yeah, lines 581-583, if the new baseFee would result in the total fee going over the max, then it limits the baseFee to the max allowed